### ISSUE TITLE
`RULInverseWeighted` does not work on `TimeSeriesWindowTransformer`

### ISSUE THREAD

**Issue by FrizzoDavide** [lucianolorenti/ceruleo#38]
I am trying to use the `sample_weight` input parameter of a  `TimeSeriesWindowTransformer` inside a `CeruleoRegressor` in order to weight the training samples with the inverse of the `RUL` in this way:

``` python
regressor = CeruleoRegressor(
			TimeSeriesWindowTransformer(
			transformer,
			window_size=1,
			sample_weight=RULInverseWeighted(),
			padding=True,
			step=1),
			Ridge(alpha=15))

regressor.fit(train_dataset)
```

Unfortunately I get a `KeyError`. The main part of the error regards the definition of the `RULInverseWeighted` class: 

``` python
class RULInverseWeighted(AbstractSampleWeights):
	"""
	Weight each sample by the inverse of the RUL
	"""
	def __call__(self, y, i: int, metadata):
		return 1 / (y[i, 0] + 1)
```

In particular there is a `KeyError: (0,0)` so it seems like the index `[i,0]` to which it is trying to access in order to compute the `RUL` of a certain sample does not exist. 

The same error is also raised in case I create a `WindowedDatasetIterator` and try to inspect its elements with the `next()` method:

``` python
iterator = WindowedDatasetIterator(
			transformed_dataset,
			window_size=150,
			step=15,
			horizon=5,
			sample_weight=RULInverseWeighted(),
			iteration_type=IterationType.FORECAST 
		)

X,y,sw=next(iterator)
(X.shape, y.shape, sw.shape)
```

**Comment by lucianolorenti** [comment]
Hi 
thank you for pointing this out.
I released a new version v3.0.4.
I think it's fixed now

**Comment by FrizzoDavide** [comment]
Ok now it works. Thanks for the help
